src/borg/archiver.py

```diff
@@ -1392,23 +1392,37 @@ def _delete_repository(self, args, repository):
 
         if not args.cache_only:
             if args.forced == 0:  # without --force, we let the user see the archives list and confirm.
+                id = bin_to_hex(repository.id)
+                location = repository._location.canonical_path()
                 msg = []
                 try:
                     manifest, key = Manifest.load(repository, Manifest.NO_OPERATION_CHECK)
+                    n_archives = len(manifest.archives)
+                    msg.append(f"You requested to completely DELETE the following repository "
+                               f"*including* {n_archives} archives it contains:")
                 except NoManifestError:
```

Comment:
if `Manifest.load()` raises this exception, `manifest` will be undefined.

Reply:
Yes, if that will fail, we will land in exception handling, i.e.
```python
                    n_archives = len(manifest.archives)
                    msg.append(f"You requested to completely DELETE the following repository "
                               f"*including* {n_archives} archives it contains:")
```
won't get executed anymore. Am I wrong?

Reply:
correct.

you could also move that `n_archives = None` to the exception handler block.